addons/purchase/models/purchase.py

```diff
@@ -190,6 +190,9 @@ def _compute_date_planned(self):
 
     @api.depends('name', 'partner_ref')
     def name_get(self):
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
```

Comment:
The `browse` doesn't fetch fields, it is the first cache miss (`po.name`).

Also I see that there are `po.amount_total` and `po.currency_id` when the context `show_total_amount` is Truly => 
actually it is a deadcode we can remove it (context never use) (your code will have more sense).